### PR TITLE
fix(codex): rewrite memory write via PreToolUse allow to stop double exec

### PR DESCRIPTION
**Pull request by efenocchi**
## Problem (F3)

A Codex memory write (`echo '...' > <memory-mount>/<dir>/file.md`) surfaced a spurious `/bin/bash: ... No such file or directory` even though the SQL write succeeded.

**Root cause — double execution.** The write-redirect branch returned plain text + exit 0. Codex >= 0.136 treats a PreToolUse exit-0 with no JSON as "allow", so it re-runs the *original* redirect on the host shell against the on-disk mount; when the subdir doesn't exist on disk the host `echo` errors. The VFS/SQL write had already happened in-hook, so the user sees a failure on a write that actually worked.

## Fix

The write-redirect branch now returns `action: "allow"` and the hook emits the PreToolUse JSON Codex honors:

```json
{"hookSpecificOutput":{"hookEventName":"PreToolUse","permissionDecision":"allow","updatedInput":{"command":"printf '%s\\n' '(done)'"}}}
```

Codex runs the harmless replacement instead of the original redirect — no double execution, no spurious error, no on-disk shadow file. The dead `guide` action (its only producer) is removed.

The contract was verified against the installed Codex (0.136.0): `codex-rs/hooks/src/events/pre_tool_use.rs` test `permission_decision_allow_can_update_input` (tag `rust-v0.136.0`) proves this exact JSON rewrites the Bash command.

## Tests

`tests/codex/codex-pre-tool-use-branches.test.ts`: allow-shape on success, POSIX single-quote escaping, negative assertions that the rewrite contains neither a redirect operator nor the memory path (proves no host re-execution), and failure -> block fallback. Full codex suite green (180/180); tsc clean.

## Not verified

Live `codex exec` round-trip — gated on model credentials + non-prod tables (same gate as the original cross-agent pass). Covered by source-contract + unit + bundle checks.

## Summary by CodeRabbit

* **New Features**
  * Updated Codex pre-tool-use permissions to add an explicit **“allow”** path for safe memory write-redirect handling, including host command rewriting to prevent unintended re-execution.

* **Bug Fixes**
  * Refined redirect detection so only safe memory-write patterns are eligible, while file-descriptor redirects are handled as unsupported.

* **Tests**
  * Expanded Vitest branch coverage for memory write-redirects and additional `ls`/`find` and fallback behaviors, including success, failure, and quoting/escaping edge cases.

* **Chores**
  * Increased per-file coverage thresholds for the pre-tool-use hook.

### CODE REVIEW

File: src/hooks/codex/pre-tool-use.ts [hunk 1]
```diff
@@ -357,33 +382,34 @@ export async function processCodexPreToolUse(
...
+  //   "allow" (exit 0 + updatedInput) — ONLY for write-redirect patterns
+  //     (echo/printf/tee … > /file). The VFS write already happened above, so we
+  //     rewrite the host command to a harmless echo: Codex reports success AND
+  //     never re-runs the original redirect against the on-disk mount (F3).
+  //   "block" (exit 2) — everything else (pipes, finds, reads) to prevent host
+  //     execution and inject the result/guidance via stderr.
+  // The write-redirect guard stays narrow: a generic ">>" check would match mixed
+  // commands like `sort /etc/passwd > /vfs/out`. Anchoring to echo/printf/tee keeps
+  // the rewrite to pure-output commands whose side effect is only the SQL write.
   const isWriteRedirect = /^\s*(echo|printf|tee)\b/.test(rewritten) && /\s>>?\s/.test(rewritten);
```
[the remark this answers is not in the document]
Comment (efenocchi) [reply src/hooks/codex/pre-tool-use.ts]: Applied the relaxation in c288eb39: the redirect test is now `/(^|[^0-9&>])>>?/`, whitespace-agnostic and still excluding fd redirects (`2>`, `&>`). Added two tests — `echo 'hello'>file` (no-space) routes to allow, and `echo ... 2>file` stays on block. Good catch.

**Review by khustup2 — APPROVED** [review]
